modules/sitemaps/sitemaps-provider-decorator.php

```diff
@@ -65,6 +65,19 @@ public function get_url_list( $page_num, $object_subtype = '' ) {
 	 * @return int Total number of pages.
 	 */
 	public function get_max_num_pages( $object_subtype = '' ) {
+		switch ( $this->provider->name ) {
+			case 'posts':
+				if ( ! PLL()->model->is_translated_post_type( $object_subtype ) && PLL()->model->options['default_lang'] !== $this->lang ) {
```

Comment:
Maybe we can pass the list of translated object subtypes as an argument of the `PLL_Sitemaps_Provider_Decorator` constructor [here](https://github.com/polylang/polylang/blob/6226a8f29b1be571cf890fc73d537c2cca025e24/modules/sitemaps/sitemaps.php#L118) ?